scvi/train/_trainingplans.py

```
@@ -1050,6 +1086,11 @@ def forward(self, *args, **kwargs):
         """Passthrough to the module's forward function."""
         return self.module(*args, **kwargs)
 
+    def compute_and_log_metrics(
+        self, loss_output: LossOutput, metrics: Dict[str, ElboMetric], mode: str
+    ):
+        assert loss_output.classification != {}
```

let's raise a proper error here